packages/@aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts

```diff
@@ -48,13 +73,29 @@ export class NetworkListener extends BaseListener implements INetworkListener {
    */
   private readonly loadBalancer: INetworkLoadBalancer;
 
+  /**
+   * Protocol assigned to listener
+   */
+  private readonly protocol: Protocol;
+
+  /**
+   * Certificates array
+   */
+  private readonly certificates?: INetworkListenerCertificateProps[];
+
   constructor(scope: cdk.Construct, id: string, props: NetworkListenerProps) {
     super(scope, id, {
       loadBalancerArn: props.loadBalancer.loadBalancerArn,
-      protocol: Protocol.Tcp,
+      protocol: props.protocol || Protocol.Tcp,
       port: props.port,
+      sslPolicy: props.sslPolicy,
+      certificates: props.certificates
     });
 
+    this.protocol = props.protocol || Protocol.Tcp;
```

Comment:
Please do this analysis only once. Put it in a local variable before the `super()` call, and then reuse in both locations.

I would prefer the following:

* Default protocol to TLS if it looks like the user wants to do TLS (either certificates or ssl policy are supplied)
* Validate that the user didn't select an invalid configuration by selecting both TCP protocol *and* supplying either certificates or an ssl policy.

It's okay to throw an `Error`.